src/test/java/DirectionsServletTest.java

```diff
@@ -76,38 +80,24 @@ public void setUp() throws IOException {
     stringWriter = new StringWriter();
     printWriter = new PrintWriter(stringWriter);
     gson = new Gson();
-    /* HOLD
-    Mockito.when(directionsClientFactory.getDirectionsClient(API_KEY))
-        .thenReturn(directionsClient);
-    */
-    Mockito.when(directionsClientFactory.getDirectionsClient(Mockito.any()))
-        .thenReturn(directionsClient);
+    Mockito.when(directionsClientFactory.getDirectionsClient(API_KEY)).thenReturn(directionsClient);
     Mockito.when(response.getWriter()).thenReturn(printWriter);
   }
 
   @Test
   public void aToBWithWaypoints() throws DirectionsException, ServletException {
-    // Get optimized route of travel from Windsor, ON to Montreal, QC with waypoints
-    // Markham, ON,
-    // Waterloo, ON and Quebec City, QC in between.
-    // Should return four legs: Windsor, ON -> Waterloo, ON -> Markham, ON -> Quebec
-    // City, QC ->
-    // Montreal, QC.
-    /* HOLD
-    Mockito.when(
-            directionsClient.getDirections(
-                "A", "B", Arrays.asList("C", "D")))
-        .thenReturn(A_TO_B_WITH_WAYPOINTS);
-    */
-    Mockito.when(
-            directionsClient.getDirections(
-                Mockito.anyString(), Mockito.anyString(), Mockito.anyListOf(String.class)))
+    PowerMockito.mockStatic(AddressUtility.class);
```

Comment:
I am not familiar with `PowerMockito`, so I don't know what magic it does, but I would be concerned about:
 - Is the mock reset before every test?
 - Is the class restored to its original implementation after the test?
 - How does it handle tests that run in parallel?

In general, "static" means "the implementation is the same for everyone", and if you start re-writing the class, you are just asking for really subtle bugs.

Even if `PowerMockito` gets all this right, it's still a non-standard solution, and it's going to force anyone reading this code to answer those questions for themselves.

Reply:
Just going to echo our offline discussion here - thanks for pointing this out. `PowerMockito` does seem pretty powerful and it's interesting to know what tradeoffs are made behind all this magic.

Reply:
`RESOLVED`